src/components/application_manager/include/application_manager/command_factory.h

```diff
@@ -54,6 +58,101 @@ class CommandFactory {
   virtual CommandSharedPtr CreateCommand(
       const commands::MessageSharedPtr& message,
       commands::Command::CommandSource source) = 0;
+  virtual bool IsAbleToProcess(
+      const int32_t,
+      const application_manager::commands::Command::CommandSource) const = 0;
+};
+
+class ICommandCreator {
+ public:
+  virtual ~ICommandCreator() {}
+  virtual bool isAble() const = 0;
+  virtual CommandSharedPtr create(
+      const commands::MessageSharedPtr& message) const = 0;
+};
+
+template <typename CommandType>
+class CommandCreator : public ICommandCreator {
```

Comment:
@BSolonenko 
rename ICommandCreator -> CommandCreator.
CommandCreator -> DefaultCommandCreator

Reply:
@LuxoftAKutsan  fixed in 75224d1